File: MyMedTimer/Models/DoseLog.swift
import Foundation
import SwiftData

@Model
final class DoseLog {
    var id: UUID = UUID()
    var scheduledTime: Date = Date()
    var actualTime: Date? = nil
    var status: String = "taken"
    var medication: Medication?

    init(scheduledTime: Date, status: String = "taken", actualTime: Date? = nil) {
        self.scheduledTime = scheduledTime
        self.status = status
        self.actualTime = actualTime
    }
}

File: android/core/domain/src/main/kotlin/com/nateb/mymedtimer/domain/model/DoseLog.kt
package com.nateb.mymedtimer.domain.model

import java.time.Instant

data class DoseLog(
    val id: String,
    val scheduledTime: Instant,
    val actualTime: Instant?,
    val status: String  // "taken", "skipped", "snoozed"
)
